force-app/main/default/lwc/appFieldSet/appFieldSet.js

```diff
@@ -0,0 +1,157 @@
+ * @date    2020-04-18
+ */
+import {LightningElement, track, wire, api} from 'lwc';
+import {ShowToastEvent} from "lightning/platformShowToastEvent";
+//Utils
+import {getOrgNamespace} from 'c/appUtils';
+//Apex methods
+import getFieldSetFields from '@salesforce/apex/AppFieldSetController.getFieldSetFields';
+import insertSObject from '@salesforce/apex/AppFieldSetController.insertSObject';
+
+export default class AppFieldSet extends LightningElement {
+	isFirstRender = false;
+	error;
+	errorMessage;
+	stack;
+
+	// Design time attribute
+	@api fieldSetObject;
+	@api fieldSetApiName;
+	@api isUsedInCommunityBuilder = false;
+
+	namespace;
+	fieldSetData;
+	@track picklistFields;
+	@track multiPicklistFields;
+	@track checkboxFields;
+	@track inputFields;
+
+	@api
+	isValid() {
+		return this.validateForm();
+	}
+
+	@api
+	getFormInputObject() {
+		return this.buildObjectToInsert();
+	}
+
+	renderedCallback() {
+		if (!this.isFirstRender) {
+			this.isFirstRender = true;
+			getOrgNamespace().then(result => {
+				if (result) {
+					this.namespace = result+'__';
+				} else {
+					this.namespace = '';
+				}
+			}).finally(() => {
+				this.loadFieldSetFields();
+			});
+		}
+	}
+
+	errorCallback(error, stack) {
+		this.error = error;
+		this.stack = stack;
+		console.error('CMP ERROR', error);
+		console.error('CMP STACK', stack);
+
+		if (error.message) {
+			this.errorMessage = error.message;
+		}
+		else if (error.body && error.body.message) {
+			this.errorMessage = error.body.message;
+		}
+		else {
+			this.errorMessage = JSON.stringify(error);
+		}
+
+		this.handleToastMessage('Error', this.errorMessage, 'error');
+	}
+
+	//Callback methods
+	loadFieldSetFields() {
+		this.template.querySelector("c-app-spinner").displaySpinner(true);
+		getFieldSetFields({fieldSetName: this.namespace+this.fieldSetApiName, ObjectName: this.namespace+this.fieldSetObject}).then(result => {
+			this.fieldSetData = result;
+		}).catch(error => {
+				this.error = error;
+				console.error('ERROR', error);
+				this.handleToastMessage(error.statusText, error.body.message, 'error');
+			}).finally(() => {
+			this.template.querySelector("c-app-spinner").displaySpinner(false);
+		});
+	}
+
+	saveAndValidate() {
+		if (this.validateForm()) {
+			this.template.querySelector("c-app-spinner").displaySpinner(true);
+			this.buildObjectToInsert();
+			let promise = this.insertObject();
+			promise.catch(error => {
+				console.error('SAVE ERROR: ', error);
+				this.handleToastMessage(error.statusText, error.body.message, 'error');
+			}).finally(() => {
+				this.template.querySelector("c-app-spinner").displaySpinner(false);
+			});
+		}
+	}
+
+	insertObject() {
+		return insertSObject({jSONSObject: JSON.stringify(this.buildObjectToInsert()), sObjectApiName: this.fieldSetObject}).then(() => {
+			this.handleToastMessage('Success', 'Successfully Inserted', 'success');
+		})
+	}
+
+	//Support methods
+	buildObjectToInsert() {
```

Comment:
loooks great! how can we support namespaced field?

Reply:
I think we should make this more generic like getRecord()?

Reply:
it builds based on the fields we get from the backend, it should handle namespace already